### ISSUE TITLE
Use unquote PATH_INFO with wsgi/python

### ISSUE THREAD

**Issue by avkarenow** [phusion/passenger#1828]
Hello,

next problem is with enoding PATH_INFO when python is used:
url: http://HOST/url%20with%20space
Passenger: PATH_INFO: /url%20with%20space
Nginx + FastCGI: PATH_INFO: /url with space
Django runserver: 'PATH_INFO': u'/url with space',
PHP: _SERVER["PATH_INFO"]	/url with space

Testing APP:
```python
#!/usr/bin/env python

def application(environ, start_response):
        start_response("200 OK", [])
        return environ["PATH_INFO"]
```

Patch to fix this attached.
[helper-scripts__wsgi-loader.py.patch2.txt](https://github.com/phusion/passenger/files/316432/helper-scripts__wsgi-loader.py.patch2.txt)

**Comment by OnixGH** [comment]
Thanks, it looks like the requirements for Python are opposite of what we need for Rails. We'll have a look at fixing this; the best place would be in the C++ Core instead of the loader though.

**Comment by sebastianlipponer** [comment]
Because of this issue it is not possible to properly run Seafile's Django web-ui (seahub) using Apache/Passenger.

Although quite ugly, the proposed wsgi-loader modification works. Please fix it in the C++ Core asap.

**Comment by OnixGH** [comment]
@sebastianlipponer I have Seafile (incl. web ui) 6.0.4 running successfully with Nginx/Passenger 5.0.30.

**Comment by sebastianlipponer** [comment]
@OnixGH Yes, the web-ui runs, but not properly. Without the fix it is not possible to access files in folders with whitespaces in their names.

**Comment by CamJN** [comment]
Looks like someone translated this patch to apply to seafile directly too. We'll see if that gets landed.

**Comment by avkarenow** [comment]
Is this issue fixed or not yet?

**Comment by CamJN** [comment]
Not yet, sorry.

**Comment by samupl** [comment]
Are there any plans on working on this issue?
